internal/controller/operator/factory/reconcile/statefulset.go

```diff
@@ -252,7 +252,7 @@ func performRollingUpdateOnSts(ctx context.Context, podMustRecreate bool, rclien
 	for _, pod := range podsForUpdate {
 		l.Info("updating pod", "pod", pod.Name)
 		// we have to delete pod and wait for it readiness
-		err := rclient.Delete(ctx, &pod, &client.DeleteOptions{GracePeriodSeconds: ptr.To(int64(30))})
+		err := rclient.Delete(ctx, &pod, &client.DeleteOptions{GracePeriodSeconds: terminationGracePeriodSeconds})
```

Comment:
I think it's better to remove `client.DeleteOptions` from rclient.Delete args.
According to the documentation, it's only required to be set if lower period compared to the value defined at pod level is needed.

 It should solve an issue with ignored `terminationGracePeriodSeconds` for all statefulset components.

Reply:
sounds good to me. just pushed those changes up. let me know how it looks